source/rock/middle/TypeDecl.ooc

```diff
@@ -463,6 +464,43 @@ TypeDecl: abstract class extends Declaration {
 
     }
 
+    checkFinalInherit: func(res: Resolver) -> Bool{
+        list := ArrayList<TypeDecl> new()
+        current := this
+
+        while(current != null) {
+            if(current getSuperType() == null) break // it's alright
+
+            next := current getSuperRef()
+            if(next == null) {
+                res wholeAgain(this, "need superRef to check final redefine")
+                return false
+            }
+
+            list add(current)
+            current = next
+        }
+
+        if(list size > 2){
+            for(i in 0..list size - 1){
+                for(j in i+1..list size){
+                    list[i] functions each(|fdecl|
+                        if(fdecl name == "init" || fdecl name == "new"){ return }
```

Comment:
I'm still not sure I understand. `init: final func` doesn't make much sense to me, but with this patch, it seems to me the `final` keyword in a `init: final func` will be ignored and have no effect. Wouldn't it make more sense to disallow `final` in this situation, rather than ignoring it?